app/assets/javascripts/discourse/app/components/topic-map/topic-map-link.gjs

```diff
@@ -0,0 +1,42 @@
+import Component from "@glimmer/component";
+import replaceEmoji from "discourse/helpers/replace-emoji";
+import and from "truth-helpers/helpers/and";
+
+export default class TopicMapLink extends Component {
+  get linkClasses() {
+    return this.args.attachment
+      ? "topic-link track-link attachment"
+      : "topic-link track-link";
+  }
+
+  get truncatedContent() {
+    const truncateLength = 85;
+    const content = this.args.title || this.args.url;
+    return content.length > truncateLength
+      ? `${content.slice(0, truncateLength).trim()}...`
+      : content;
+  }
+
+  <template>
+    <a
+      class={{this.linkClasses}}
+      href={{@url}}
```

Comment:
Is this going to handle subfolders?

Reply:
In this case, they are treated like any other link, so it should be fine. It's the same component we have been using for a while too